CDCgov/prime-simplereport
# Bob/5817 return focus on archive close

# FRONTEND PULL REQUEST

## Related Issue
Fixes #5817 

## Changes Proposed

- Returns focus to the action menu of the relevant user when the archive modal is closed
- Adds a corresponding test
- Retain the default focus to the patient search bar 
- Decided to refocus on the action menu for the user row in question rather than the explicit archive button, since 
   - I'm guessing that when users go back from the archive menu, moving back to the action menu would allow them to redirect themselves as needed (ie tab to the next user row, move to the "Start test" option instead) better than the actual archive button in the expanded list.
   - The secondary action dropdown isn't expanded on modal exit and it didn't seem worth it to expand it programmatically in this case

## Testing

- deployed on dev 2
- To replicate
    - Navigate to the manage patients menu
    - Select archive a patient
    - Exit out of the modal and notice focus returns to the action button

## Screenshots / Demos
https://github.com/CDCgov/prime-simplereport/assets/29645040/75d507f4-250f-437d-b9d3-021153013123


## Checklist for Author and Reviewer

### Accessibility
- [x] Any large changes have been run through Deque manual testing
- [x] All changes have run through the Deque automated testing
